tools/remove_from.hpp

```diff
@@ -12,63 +12,60 @@
 namespace antler {
    struct remove_from_project {
 
-      template <antler::project::object::type_t Ty>
+      template <typename Obj>
       bool remove_obj(antler::project::project& proj) {
          if (obj_name.empty()) {
-            std::cerr << "Name is empty" << std::endl;
+            system::error_log("Object name is empty.");
             return false;
          }
 
-         if (proj.remove(obj_name, Ty)) {
-            std::cout << "Removing object: " << obj_name << " from the project." << std::endl;
+         if (proj.remove(obj_name)) {
+            system::info_log("Object {0} removed from the project.", obj_name);
             return true;
          }
          return false;
       }
 
-      inline bool remove_app(antler::project::project& proj) { return remove_obj<antler::project::object::type_t::app>(proj); }
-      inline bool remove_lib(antler::project::project& proj) { return remove_obj<antler::project::object::type_t::lib>(proj); }
-      inline bool remove_test(antler::project::project& proj) { return remove_obj<antler::project::object::type_t::test>(proj); }
+      inline bool remove_app(antler::project::project& proj) { return remove_obj<antler::project::app_t>(proj); }
+      inline bool remove_lib(antler::project::project& proj) { return remove_obj<antler::project::lib_t>(proj); }
+      //inline bool remove_test(antler::project::project& proj) { return remove_obj<antler::project::test_t>(proj); }
 
-      bool remove_dependency(antler::project::project& proj) {
-         if (!dep_name.empty()) {
-            if (!obj_name.empty()) {
-               // Get the object to operate on.
-               try {
-                  auto& obj = proj.object(obj_name);
-
-                  if (obj.remove_dependency(dep_name)) {
-                     std::cout << "Removing dependency: " << dep_name << " from: " << obj_name << std::endl;
-                     return true;
-                  } else {
-                     std::cerr << "Dependency: " << dep_name << " is not a dependency of: " << obj_name << std::endl;
-                     return false;
-                  }
-               } catch(...) {
-                  std::cerr << "Object: " << obj_name << " does not exist." << std::endl;
+      bool remove_dependency_from_all(antler::project::project& proj) {
+         const auto& remove_dep = [](auto dep, auto& objs) {
+            for (auto& [k, o] : objs) {
+               if (o.remove_dependency(dep)) {
+                  std::cout << "Removing dependency: " << dep << " from: " << o.name() << std::endl;

```

Comment:
I assume system::info_log supposed to be used here

Reply:
Fixed in c1f01273